docker/Dockerfile.fedora

```diff
@@ -0,0 +1,30 @@
+# This Dockerfile is used to both document and test building bpftrace
+# on the latest fedora. We would use `fedora:latest`, but to make the
+# CI more deterministic, we pin to latest release.
```

Comment:
I'd argue that using `fedora:latest` is what we want here b/c it will allow CI to catch build problems very early. Maybe we could go even further and use `fedora:rawhide` to try to build against the upcoming release.

Reply:
Rawhide sounds reasonable. Given unstable nature of living at tip (same with `:latest`), I'd probably wanna only run the distro CI post-merge (IOW on master). WDYT?

Reply:
Yes, good idea, that'll notify us about a distro breakage and not break PRs at the same time.